src/subscription/data/details/sagas.js

```diff
@@ -0,0 +1,177 @@
+//   }
+// }
+
+export function* handleFetchSubscriptionDetails() {
+  if (yield isSubscriptionDetailsProcessing()) {
+    // Do nothing if there is a request currently in flight to get or modify the details
+    return;
+  }
+
+  try {
+    yield put(subscriptionDetailsProcessing(true)); // we are going to modify the details, don't make changes
+    const result = yield call(SubscriptionApiService.getDetails);
+    yield put(subscriptionDetailsReceived(result)); // update redux store with details data
+    yield call(handleMessages, result.messages, true, window.location.search);
+    // yield call(handleDiscountCheck); // check if a discount should be added to the details
+  } catch (error) {
+    yield call(handleErrors, error, true);
+    if (error.details) {
+      yield put(subscriptionDetailsReceived(error.details)); // update redux store with details data
+      // yield call(handleDiscountCheck); // check if a discount should be added to the details
+    }
+  } finally {
+    yield put(subscriptionDetailsProcessing(false)); // we are done modifying the details
+    yield put(fetchSubscriptionDetails.fulfill()); // mark the details as finished loading
+  }
+}
+
+/**
+ * Many of the handlers here have identical error handling, and are also all processing the same
+ * sort of response object (a details).  This helper is just here to dedupe that code, since its
+ * all identical.
+ */
+export function* performSubscriptionDetailsOperation(operation, ...operationArgs) {
+  if (yield isSubscriptionDetailsProcessing()) {
+    return;
+  }
+
+  try {
+    yield put(subscriptionDetailsProcessing(true));
+    const result = yield call(operation, ...operationArgs);
+    yield put(subscriptionDetailsReceived(result));
+    yield call(handleMessages, result.messages, true, window.location.search);
+  } catch (error) {
+    yield call(handleErrors, error, true);
+    if (error.details) {
+      yield put(subscriptionDetailsReceived(error.details));
+    }
+  } finally {
+    yield put(subscriptionDetailsProcessing(false));
+  }
+}
+
+// export function* handleAddCoupon({ payload }) {
+//   yield call(performSubscriptionDetailsOperation, SubscriptionApiService.postCoupon, payload.code);
+// }
+
+// export function* handleRemoveCoupon({ payload }) {
+//   yield call(performSubscriptionDetailsOperation, SubscriptionApiService.deleteCoupon, payload.id);
+// }
+
+// export function* handleUpdateQuantity({ payload }) {
+//   yield call(performSubscriptionDetailsOperation, SubscriptionApiService.postQuantity, payload);
+// }
+
+export function* handleSubmitPayment({ payload }) {
+  if (yield isSubscriptionDetailsProcessing()) {
+    return;
+  }
+
+  const { method, ...paymentArgs } = payload;
+  try {
+    yield put(subscriptionDetailsProcessing(true));
+    yield put(clearMessages()); // Don't leave messages floating on the page after clicking submit
+    yield put(submitPayment.request());
+    const paymentMethodCheckout = paymentMethods[method];
```

Comment:
As commented in another place, we'll only have 1 payment method, so no need for this logic to map out the method to the respective checkout function

Reply:
updated to use `subscriptionStripeCheckout` method but leave the implementation as it is, so we can easily add more methods in future. 
```
import * as SubscriptionApiService from '../service';
import { subscriptionStripeCheckout } from '../../subscription-methods';

export const paymentMethods = {
  stripe: subscriptionStripeCheckout,
};
```

Reply:
Sounds good

Reply:
Sounds good!